commit ed4a5208840986145a135e634c1c457fcd45c80e (refs/changes/49/4149/7)
Author: Brian Bennett <brian.bennett@joyent.com>
Date:   2018-06-12T16:39:02-07:00 (1 year, 4 months ago)
    
    MANATEE-397 `manatee-adm history` fails with `TypeError: Cannot read property 'zoneId' of null` (again)
    Reviewed by: Josh Wilsdon <josh@wilsdon.ca>

diff --git a/lib/adm.js b/lib/adm.js
index 92b2636..275c450 100644
--- a/lib/adm.js
+++ b/lib/adm.js
@@ -1927,6 +1927,17 @@ function annotateHistoryNode(evt, last)
     }
 
     if (nst.generation > lst.generation) {
+        if (lst.sync === null && lst.oneNodeWriteMode !== true) {
+            /*
+             * We seem to have arrived here due to an invalid state on the
+             * previous generation, namely the previous generation is
+             * multi-mode, writable, with no replicas. This should never
+             * happen, but somehow it did. At any rate, we're just happy that
+             * there's a replica now.
+             */
+            return (sprintf('sync "%s" added',
+                nst.sync.zoneId.substr(0, 8)));
+        }
         if (nst.sync.zoneId == lst.sync.zoneId) {
             return ('error: gen number changed, but ' +
                 'primary and sync did not');
